tests/scripts/ncp/test_ncp_mac.py

```diff
@@ -0,0 +1,93 @@
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import os
+import sys
+import time
+import pexpect
+import unittest
+
+import node
+
+class test_ncp_mac(unittest.TestCase):
+    def setUp(self):
+        self.node = node.Node(1)
+
+    def test_mac_saddr(self):
+        self.node.send_command('ncp-mac16')
+        self.node.pexpect.expect('Done')
+
+    def test_mac_laddr(self):
+        self.node.send_command('ncp-mac64')
+        self.node.pexpect.expect('Done')
+
+    def test_mac_filter_mode(self):
+        self.node.send_command('mac-filter-mode')
+        self.node.pexpect.expect('Done')
+
+        self.node.send_command('mac-filter-mode 1')
+        self.node.pexpect.expect('Done')
+
+        self.node.send_command('mac-filter-mode 2')
+        self.node.pexpect.expect('Done')
+
+        self.node.send_command('mac-filter-mode 0')
+        self.node.pexpect.expect('Done')
+        self.node.send_command('mac-filter-mode')
+        self.node.pexpect.expect('0')
+        self.node.pexpect.expect('Done')
+
+    def test_hwaddr(self):
+        self.node.send_command('ncp-eui64')
+        self.node.pexpect.expect('Done')
+
+    def test_scan(self):
+        self.node.send_command('scan')
+        self.node.pexpect.expect('Done')
+
+    def test_whitelist(self):
+        self.node.send_command('whitelist')
+        self.node.pexpect.expect('Disabled')
+        self.node.pexpect.expect('Done')
+
+        self.node.send_command('whitelist enable')
+        self.node.pexpect.expect('Done')
+
+        self.node.send_command('whitelist')
+        self.node.pexpect.expect('Enabled')
+        self.node.pexpect.expect('Done')
+
+        self.node.send_command('whitelist add deadbeeff00dbabe')
```

Comment:
Check again, `remove` is a part of the current test suite.  You are probably not seeing coverage in NCP due to #517.